packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -99,3 +100,27 @@ bool debugCheckHasTable(BuildContext context) {
   });
   return true;
 }
+
+void debugWidgetBuilderValue(Widget widget, Widget built) {
+  assert(() {
+    if (built == null) {
+      throw new FlutterError(
+        'A build function returned null.\n'
+        'The offending widget is: $widget\n'
+        'Build functions must never return null. '
+        'To return an empty space that causes the building widget to fill available room, return "new Container()". '
+        'To return an empty space that takes as little room as possible, return "new Container(width: 0.0, height: 0.0)".'
+      );
+    }
+    return true;
+  });
+}
+
+void debugReportException(String context, dynamic exception, StackTrace stack) {
```

Comment:
I'd rather we not make this public. That is, I'd rather you duplicate the call to reportError() that have a public function that hard-codes the "widgets library" string. The reason is that otherwise people will call this from their app and then get error messages that say that the "widgets library" caught their exception.